test/BlockChainHelper.cpp

```diff
@@ -0,0 +1,396 @@
+
+	if (_tObj.count("parentHash"))
+		rlpStream << importByteArray(_tObj.at("parentHash").get_str());
+
+	if (_tObj.count("uncleHash"))
+		rlpStream << importByteArray(_tObj.at("uncleHash").get_str());
+
+	if (_tObj.count("coinbase"))
+		rlpStream << importByteArray(_tObj.at("coinbase").get_str());
+
+	if (_stateRoot)
+		rlpStream << _stateRoot;
+	else if (_tObj.count("stateRoot"))
+		rlpStream << importByteArray(_tObj.at("stateRoot").get_str());
+
+	if (_tObj.count("transactionsTrie"))
+		rlpStream << importByteArray(_tObj.at("transactionsTrie").get_str());
+
+	if (_tObj.count("receiptTrie"))
+		rlpStream << importByteArray(_tObj.at("receiptTrie").get_str());
+
+	if (_tObj.count("bloom"))
+		rlpStream << importByteArray(_tObj.at("bloom").get_str());
+
+	if (_tObj.count("difficulty"))
+		rlpStream << bigint(_tObj.at("difficulty").get_str());
+
+	if (_tObj.count("number"))
+		rlpStream << bigint(_tObj.at("number").get_str());
+
+	if (_tObj.count("gasLimit"))
+		rlpStream << bigint(_tObj.at("gasLimit").get_str());
+
+	if (_tObj.count("gasUsed"))
+		rlpStream << bigint(_tObj.at("gasUsed").get_str());
+
+	if (_tObj.count("timestamp"))
+		rlpStream << bigint(_tObj.at("timestamp").get_str());
+
+	if (_tObj.count("extraData"))
+		rlpStream << fromHex(_tObj.at("extraData").get_str());
+
+	if (_tObj.count("mixHash"))
+		rlpStream << importByteArray(_tObj.at("mixHash").get_str());
+
+	if (_tObj.count("nonce"))
+		rlpStream << importByteArray(_tObj.at("nonce").get_str());
+
+	return rlpStream.out();
+}
+
+//Form bytestream of a block with [header transactions uncles]
+void TestBlock::recalcBlockHeaderBytes(RecalcBlockHeader _recalculate)
+{
+	Transactions txList;
+	for (auto const& txi: m_transactionQueue.topTransactions(std::numeric_limits<unsigned>::max()))
+		txList.push_back(txi);
+	RLPStream txStream;
+	txStream.appendList(txList.size());
+	for (unsigned i = 0; i < txList.size(); ++i)
+	{
+		RLPStream txrlp;
+		txList[i].streamRLP(txrlp);
+		txStream.appendRaw(txrlp.out());
+	}
+
+	RLPStream uncleStream;
+	uncleStream.appendList(m_uncles.size());
+	for (unsigned i = 0; i < m_uncles.size(); ++i)
+	{
+		RLPStream uncleRlp;
+		m_uncles[i].getBlockHeader().streamRLP(uncleRlp);
+		uncleStream.appendRaw(uncleRlp.out());
+	}
+
+	if (_recalculate == RecalcBlockHeader::Update || _recalculate == RecalcBlockHeader::UpdateAndVerify)
+	{
+		if (m_uncles.size()) // update unclehash in case of invalid uncles
+			m_blockHeader.setSha3Uncles(sha3(uncleStream.out()));
+
+		//transactions hash??
+
+		dev::eth::mine(m_blockHeader);
+		m_blockHeader.noteDirty();
+	}
+
+	RLPStream blHeaderStream;
+	m_blockHeader.streamRLP(blHeaderStream, WithProof);
+
+	RLPStream ret(3);
+	ret.appendRaw(blHeaderStream.out()); //block header
+	ret.appendRaw(txStream.out());		 //transactions
+	ret.appendRaw(uncleStream.out());	 //uncles
+
+	if (_recalculate == RecalcBlockHeader::Verify || _recalculate == RecalcBlockHeader::UpdateAndVerify)
+		m_blockHeader.verifyInternals(&ret.out());
```

Comment:
the reason to verifyInternals is to check wether it throws or not. Why are you checking it here, or why don't you have a tr/catch around it?